### DISCUSSION TITLE
MBXP 1.3.1 Alpha

### DISCUSSION THREAD

**Discussion by tlefko** [tlefko/Matebook-X-Pro-2018-Monterey#43] [Show and tell]
**Major Release***

This is a Beta, Looking for Testers and Feedback!

- Improved Wifi Stability and Connectivity
- Universal Trackpad/Keyboard ACPIs Published as a fix
- Integrated Better CPU Management and Wifi Power Management
- Improved UI Fluidity in menus and wifi settings
- Previous Fixes Mentioned for USB/TB in previous release

Credits: @Paxolo @yonathara @0xTides @Lukas3118 @GaryYa0 @yonathanarya

<hr /><em>This discussion was created from the release <a href='https://github.com/tlefko/Huawei-Matebook-X-Pro-Ventura/releases/tag/alpha'>MBXP 1.3.1 Alpha</a>.</em>

**Comment by Paxolo** [comment]
Hi @tlefko, I read all the messages in the issue pages but wasn't able to test anything because I wasn't home.

As soon as I got back, I restarted my laptop using the new EFI and here's my experience:
Wi-Fi performance is even better now, almost doubling transfer speeds once again on my machine. At this point, it's basically the same performance I would have on other OS's.
However, internal input devices (keyboard, trackpad and touchscreen) are still broken.
Let me know if there's any file/log I can share that could provide useful information!

Edit: I did NOT perform a clean installation this time, so that might be part of the problem...

**Comment by tlefko** [comment]
Hmmm, well that's annoying to hear lol. Here, run this and send me the
output.

Should be able to email to l e f k o t y l e r * g m a i l . c o m Hopefully I can find some
clues. Glad to hear wifi is improved and like other OS!

https://drive.google.com/file/d/1PflM9l74n-zahtkuohXz_PtJR0v0FBbF/view?usp=sharing

**Reply by Paxolo** [reply]
I ran the tool and emailed you the output. Hope you find it helpful!

**Reply by tlefko** [reply]
> I ran the tool and emailed you the output. Hope you find it helpful!

Am looking through it now, will send fix to try. Not sure if this will work, it is a GPIO pin issue. If possible, if you could film the verbose output during boot and email that would be the best way to see the exact error. early in the boot it loads Voodooi2c/ps2. -- I have emailed you the EFI to try as it wont let me attach.

**Reply by Paxolo** [reply]
I tried the EFI you emailed me but peripherals are still broken.
I'll try my best to film the OS boot since text goes by super fast and I'm afraid you won't be able to read much.
If I manage to get something readable, I will upload it to YouTube as unlisted and send you a link to it.

**Reply by Paxolo** [reply]
I made the video and I'm uploading it. It'll a while as upload speeds on my phone are kinda bad...

In the meantime, though, I was wondering if there's a way to just output the verbose boot log into a file and found this.
![image](https://user-images.githubusercontent.com/53195387/229308490-8d060c4e-ab6a-48dd-9e03-cefd9376b9a8.png)

Apparently, setting target to 0x40 (hex for 64) would just write all of that verbose output to a log file (?)

Edit: 
I actually had to set it to 67 for it to actually write to a file. 
After that, the laptop wouldn't boot because of a missing Info.plist inside the BrightnessKeys kext.
So, I looked in OC/Kexts and found out there was no such kext in there.
I then prevented it from loading setting its value to False in the config.plist and now the laptop boots again.
Here's the file I obtained: 

[opencore-2023-04-01-203318.txt](https://github.com/tlefko/Huawei-Matebook-X-Pro-Ventura/files/11130547/opencore-2023-04-01-203318.txt)

**Reply by tlefko** [reply]
@Paxolo hey, so if you read through it it doesn't actually log in any of the jernel extensions loading just the OC stuff. Just try your best to film, its a really small area i need to read so i can puase it and find it its no worries. ---Never mind just saw video, watching now

**Reply by tlefko** [reply]
> I tried the EFI you emailed me but peripherals are still broken. I'll try my best to film the OS boot since text goes by super fast and I'm afraid you won't be able to read much. If I manage to get something readable, I will upload it to YouTube as unlisted and send you a link to it.
<img width="1445" alt="Screenshot 2023-04-02 at 12 43 22 PM" src="https://user-images.githubusercontent.com/42879340/229366907-e3cf2f68-70da-439f-80cc-85ee2163e4d6.png">


Ok, I have found the issue. Looks like the it cannot attach to the trackpad device like I thought.. Will try to work on a fix

**Reply by Paxolo** [reply]
Great! Let me know if there's something you want me to test

**Reply by tlefko** [reply]
> Great! Let me know if there's something you want me to test

Probably going to be inactive for a little bit, exam season coming up so I hope everything is usable for you right now. if not we can revert

**Comment by 0xTides** [comment]
@tlefko Does this release support the 2019 model? If so I will try this with a Ventura install tomorrow :)

**Reply by 0xTides** [reply]
Unfortunately, I tried updating my kexts and config.plist to the ones here. I was able to boot but the update has broken my install. Any recommendations @tlefko

**Reply by tlefko** [reply]
@0xTides Yeah, haha that would be my bad as that version did not support 2019. I believe this version should be succesful for your model and implement all the new fixes and features, please lmk. --- It is in the alpha release section now

**Reply by 0xTides** [reply]
@tlefko Thats also my bad, when I first saw the previous release I didn’t see the 2019 release. I will test on fresh Ventura install tonight :)

**Reply by 0xTides** [reply]
@tlefko Successfully on Ventura! Had some issues with finding the right Ventura installer, but managed to get it going.

Everything is smooth with only a couple of issues. The performance feels really good!

Slight issues I've experienced so far:
- wifi is slightly slow to get going and find networks. No issues when connected. - _Screen mirroring does not work due to wifi issues (my guess)_
- iCloud has slight issues (Settings menu is very slow, and signing into Apple ID via iCloud menu is very slow/times out. Signing in through the App Store fixes this. - _Fixed with update from 13.1 > 13.3_

Very impressed with the 2019 release and will provide more updates on stability as I use more. :) Thanks again

**Reply by tlefko** [reply]
@0xTides Thank you for the feedback! Will work to improve this. very unsure what's causing the second issues but could be the wifi -- hard to fix as it's not really developed by me but am doing my best

**Comment by elfzweik** [comment]
Hello tlefko, long time no see! :)
Glad you are still working on the fine improvement of this machine and have great progress. I did not touch macos for quite a period. Last week I tried 1.3.1 Alpha to test the improvements.  My report is following: 
1. BrightnessKeys.kext was not included in the release. 
2. I made a clean installation of Big Sur. Macos did not recognize Wifi and  Bluetooth modules. It was finally proved to be my mistake.  I didn't replace the AirportIwm.kext with the Big Sur version. After fixing that error, the wifi works perfect. while bluetooth cannot connect my logitek anywhere 3s mouse and logitek keys keyboard. But it works with my old logitek 380 keyboard, which reminded me that the intelBluetoothFirmware's readme said: Bluetooth 5.x module can connect to Bluetooth 4.x devices. But Bluetooth 4.x module still cannot connect to Bluetooth 4.x devices on macos older than Monterey. 
3. Update to Ventura from AppStore was proved unsuccessful. I can enter the desktop. But the booting cost so long time and it is very slow too after booting. 
4. After fresh installation of Monterey, my logitek mouse and keyboard can be connected now. But after sleep, the anywhere 3 mouse was found, while keys keyboard not. it's weird. Wifi works perfect, even after sleep. 
5. The boot is much slower than that on Big Sur. It is caused by my Samsung 980 pro again. After set SetApfsTrimTimeout to zero, the boot is much faster and acceptable. And I also updated NvmeFix.kext to 1.1.0. Not sure the shorten time caused by which factor or both.

**Reply by elfzweik** [reply]
More updates of the test:
1. More sleep tests prove that the bluetooth can recover after sleep, but it is quite slow. And the bluetooth were turned off twice in my many tests. In this case you need to turn it on manually. Again, the status refreshes very slowly. When I switched it on, it still shows off, while bluetooth devices can be connected already. The status shows correctly after 1 or 2 minutes. 
2. Airdrop works fine now.
3. The ACPI patches still cannot support windows and linux booting. I resolved this problem by add OC menu entry to the Linux's grub2 menu. Now I can select to run windows, macos or linux at boot. It works perfectly.
![](https://raw.githubusercontent.com/elfzweik/fighouse/main/2023/07/25-16-47-59-41690274718_.pic.jpg)
Thanks again, tlefco, for your great job.

**Reply by tlefko** [reply]
@Paxolo @elfzweik thank you for continued testing and support of the proejct. I genuijnely am blown away by the community we've created here and the amazing testing and creative work you guys did to problem solve

I've had one of the most insane year's of my entire life haha, just been in and out of the hospital; a lot of personal stuff and just reallty sorting my academic stuff after missing so much school.

I am extremely sorry for the incredibly long period of inactivity I've just been cleaning up the pieces haha

**I DO WANT TO CONTINUE TO SUPPORT THE PROJECT, and this is a big passion project of mine**

I apolagize for the extremely long period of inactivity and i look forward to getting back to work.

please update me on the status of both of these issues and i will begin creating files for testing.

thanks guys

**Reply by elfzweik** [reply]
@tlefko Sad to know you got health problem. Take care, Bro. The health is always the first priority and followed by academics. Take it easy. This project is already very close to perfection. This is the best hackintosh in my four laptops. Just let you know my appreciation to your work, I refused the company's offer for a new laptop and chose to keep this 5-year-old model just because of the perfect hackintosh. :) 👍
